### PR TITLE
add support for Xcode 15

### PR DESCRIPTION
**Pull request by siddarthkay**
After upgrading to Xcode 15 `make run-ios` would fail with 
```
❌ /Users/siddarthkumar/code/status-im/PR/status-mobile/ios/Pods/boost/boost/container_hash/hash.hpp:131:33: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
```

This PR fixes the issue and makes existing codebase compatible with newer Xcode. 
reference issue in react-native repo : https://github.com/facebook/react-native/issues/37748
        
#### Platforms
- iOS

status: ready

### CODE REVIEW

**Comment by siddarthkay** [comment]
Side effect of this approach is we now get ugly logs like this when running `make run-ios`

```
[StatusIm] Compiling main.m
⚠️  ld: option '-no_dead_strip_inits_and_terms' is obsolete, use '-dead_strip' instead
ld: warning: ignoring duplicate libraries: '-lc++', '-lz'
ld: warning: '/Users/siddarthkumar/Library/Developer/Xcode/DerivedData/StatusIm-btihnuzvzghnuvhfwbpfwxtxfhoy/Build/Products/Debug-iphonesimulator/Statusgo.framework/Versions/A/Statusgo[x86_64][2](go.o)' has malformed LC_DYSYMTAB, expected 117 undefined symbols to start at index 88103, found 260 undefined symbols starting at index 399
❌ duplicate symbol '_secp256k1_ec_pubkey_serialize' in
❌ duplicate symbol '_secp256k1_context_set_error_callback' in
❌ duplicate symbol '_secp256k1_nonce_function_default' in
❌ duplicate symbol '_secp256k1_ec_pubkey_create' in
❌ duplicate symbol '_secp256k1_ecdsa_signature_serialize_der' in
❌ duplicate symbol '_secp256k1_ecdsa_recoverable_signature_serialize_compact' in
❌ duplicate symbol '_secp256k1_ec_pubkey_combine' in
❌ duplicate symbol '_secp256k1_ecdsa_signature_parse_der' in
❌ duplicate symbol '_secp256k1_context_randomize' in
❌ duplicate symbol '_secp256k1_ecdsa_signature_serialize_compact' in
❌ duplicate symbol '_secp256k1_ec_privkey_tweak_add' in
❌ duplicate symbol '_secp256k1_ecdsa_recoverable_signature_parse_compact' in
❌ duplicate symbol '_secp256k1_ecdsa_sign_recoverable' in
❌ duplicate symbol '_secp256k1_context_destroy' in
❌ duplicate symbol '_secp256k1_ecdsa_verify' in
❌ duplicate symbol '_secp256k1_context_create' in
❌ duplicate symbol '_secp256k1_ec_seckey_verify' in
❌ duplicate symbol '_secp256k1_ec_pubkey_tweak_add' in
❌ duplicate symbol '_secp256k1_ecdsa_recover' in
❌ duplicate symbol '_secp256k1_context_set_illegal_callback' in
❌ duplicate symbol '_secp256k1_ec_pubkey_tweak_mul' in
❌ duplicate symbol '_secp256k1_ec_pubkey_parse' in
❌ duplicate symbol '_secp256k1_ecdsa_signature_normalize' in
❌ duplicate symbol '_secp256k1_ecdsa_sign' in
❌ duplicate symbol '_secp256k1_nonce_function_rfc6979' in
❌ duplicate symbol '_secp256k1_context_clone' in
❌ duplicate symbol '_secp256k1_ecdsa_recoverable_signature_convert' in
❌ duplicate symbol '_secp256k1_ecdsa_signature_parse_compact' in
❌ duplicate symbol '_secp256k1_ec_privkey_tweak_mul' in
[StatusIm] Running script Bundle React Native code and images
```

However the end result is 

```
** BUILD SUCCEEDED **
success Successfully built the app
2023-09-20 10:41:04.490 xcodebuild[34999:300739] Requested but did not find extension point with identifier Xcode.InterfaceBuilderBuildSupport.PlatformDefinition
info Installing "/Users/siddarthkumar/Library/Developer/Xcode/DerivedData/StatusIm-btihnuzvzghnuvhfwbpfwxtxfhoy/Build/Products/Debug-iphonesimulator/StatusIm.app"
info Launching "im.status.ethereum.debug"
success Successfully launched the app on the simulator
```

So I guess the red Xs are safe to ignore for now.

File: ios/Podfile.lock [hunk 1]
```diff
@@ -772,7 +772,7 @@ SPEC CHECKSUMS:
   RNLanguages: 962e562af0d34ab1958d89bcfdb64fafc37c513e
   RNPermissions: ad71dd4f767ec254f2cd57592fbee02afee75467
   RNReactNativeHapticFeedback: 2566b468cc8d0e7bb2f84b23adc0f4614594d071
-  RNReanimated: d0db0ee059c33381bca787a2193c27e52750ccaf
+  RNReanimated: b3b67ebe099c0b0e7b5c7386b18d2468e29c9d41
```
Comment (flexsurfer) [inline ios/Podfile.lock:775]: someone forgot to push this change?

**Review by flexsurfer — APPROVED** [review]

**Comment by flexsurfer** [comment]
i gave approve because i don't see anything wrong, but i haven't tested it

**Comment by qfrank** [comment]
trying with xcode 14.3.1

**Comment by siddarthkay** [comment]
Thanks @flexsurfer and @qfrank you read my mind. I was just now thinking about backward compatibility with Xcode 14.x :)
Although CI is green but its important for this change not to break development environment of other developers.
Looking forward to your results 👍🏻

File: ios/Podfile.lock (hunk 1 above)
Comment (siddarthkay) [reply ios/Podfile.lock:775]: I guess

**Review by qfrank — APPROVED** [review]
LGTM, xcode 14.3.1 works as before, applogize for not brave enough to try xcode 15 but still thank you for your PR ❤️

**Review by yakimant — APPROVED** [review]

**Review by mohsen-ghafouri — APPROVED** [review]
I just upgraded and tested it with Xcode 15 , all works fine. thanks :)

**Comment by siddarthkay** [comment]
@status-im/mobile-qa : merged this without QA because we're only adding support to build the app with Xcode 15 and this has no impact on app functionalities. Also E2E was run just in case ;)

**Comment by jakubgs** [comment]
A more accurate reason would be that Xcode changed cannot affect E2E tests since those are run on Android builds.

**Pull request merged**